drivers/adc/adc_stm32.c

```diff
@@ -585,6 +596,9 @@ static int adc_stm32_init(struct device *dev)
 	}
 #endif
 
+#ifdef CONFIG_SOC_SERIES_STM32WBX
+	LL_ADC_REG_SetTriggerSource(adc, LL_ADC_REG_TRIG_SOFTWARE);
```

Comment:
Same question here (why STM32WB only).
Sorry, missed it previously.

Reply:
Same answer.  It's actually unnecessary since it is in software trigger mode by default.  I removed it.